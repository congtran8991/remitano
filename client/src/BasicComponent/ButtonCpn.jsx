import React from "react";

const ButtonCpn = (props) => {
  const { title = "", onClick = () => {}, style = {}, dataTestId = "defaultTestId" } = props;
  return (
    <button
      data-testid={dataTestId}
      style={style}
      onClick={onClick}
      className="btn"
    >
      {title}
    </button>
  );
};

export default ButtonCpn;
