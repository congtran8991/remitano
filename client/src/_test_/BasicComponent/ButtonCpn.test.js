import React from "react";
import { render, fireEvent, screen } from "@testing-library/react";
import Enzyme from "enzyme";
import Adapter from "enzyme-adapter-react-16";
import ButtonCpn from "../../BasicComponent/ButtonCpn";

Enzyme.configure({ adapter: new Adapter() });
describe("ButtonCpn", () => {
  test("calls onClick handler when clicked", () => {
    render(<ButtonCpn dataTestId="defaultTestBtnId"/>)
    const btnCpn = screen.getByTestId("defaultTestBtnId");
    fireEvent.click(btnCpn)
  });
});
