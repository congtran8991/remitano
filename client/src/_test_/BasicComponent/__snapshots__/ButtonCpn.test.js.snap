// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[` 1`] = `undefined`;

exports[`ButtonCpn Render ButtonCpn 1`] = `undefined`;
